examples/fem128.py

```diff
@@ -0,0 +1,128 @@
+import taichi as ti
+try:
+    import taichi_glsl as tl
+except ImportError:
+    print('This example needs the extension library Taichi GLSL to work.'
+          'Please run `pip install --user taichi_glsl` to install it.')
+ti.init(arch=ti.gpu)
+
+N = 12
+dt = 5e-5
+dx = 1 / N
+NF = 2 * N ** 2
+NV = (N + 1) ** 2
+E, nu = 4e4, 0.2
+mu, lam = E / 2 / (1 + nu), E * nu / (1 + nu) / (1 - 2 * nu)
+ball_pos, ball_radius = tl.vec(0.5, 0.0), 0.31
+damping = 14.5
+
+pos = ti.Vector.var(2, ti.f32, NV, needs_grad=True)
+vel = ti.Vector.var(2, ti.f32, NV)
+f2v = ti.Vector.var(3, ti.i32, NF)
+B = ti.Matrix.var(2, 2, ti.f32, NF)
+F = ti.Matrix.var(2, 2, ti.f32, NF, needs_grad=True)
+V = ti.var(ti.f32, NF)
+phi = ti.var(ti.f32, NF)
+U = ti.var(ti.f32, (), needs_grad=True)
+
+gravity = ti.Vector.var(2, ti.f32, ())
+attractor_pos = ti.Vector.var(2, ti.f32, ())
+attractor_strength = ti.var(ti.f32, ())
+
+@ti.kernel
+def update_U():
+    for i in range(NF):
+        ia, ib, ic = f2v[i]
+        a = pos[ia]
+        b = pos[ib]
+        c = pos[ic]
+        V[i] = abs((a - c).cross(b - c))
+        D_i = ti.Matrix.cols([a - c, b - c])
+        F[i] = D_i @ B[i]
+    for i in range(NF):
```

Comment:
Btw, I found that removing this:
```suggestion
```
will cause blow up on contact, is using two loop expected of FEM elasticity? Or a autodiff bug?

Reply:
Hi, I've got a question here. Since V[I] is the area of the triangle element, should we multiply a 0.5 on the cross product here?